src/gs-update-monitor.c

```diff
@@ -413,19 +415,57 @@ check_thrice_daily_cb (gpointer data)
 	return G_SOURCE_CONTINUE;
 }
 
+static void
+stop_upgrades_check (GsUpdateMonitor *monitor)
+{
+	if (monitor->check_daily_id == 0)
+		return;
+
+	g_source_remove (monitor->check_daily_id);
+	monitor->check_daily_id = 0;
+}
+
+static void
+restart_upgrades_check (GsUpdateMonitor *monitor)
+{
+	stop_upgrades_check (monitor);
+	get_upgrades (monitor);
+
+	monitor->check_daily_id = g_timeout_add_seconds (3 * 86400,
+							 check_thrice_daily_cb,
+							 monitor);
+}
+
+static void
+stop_updates_check (GsUpdateMonitor *monitor)
+{
+	if (monitor->check_hourly_id == 0)
+		return;
+
+	g_source_remove (monitor->check_hourly_id);
+	monitor->check_hourly_id = 0;
+}
+
+static void
+restart_updates_check (GsUpdateMonitor *monitor)
+{
+	stop_updates_check (monitor);
+	check_updates (monitor);
+
+	monitor->check_hourly_id = g_timeout_add_seconds (3600, check_hourly_cb,
+							  monitor);
+}
+
 static gboolean
 check_updates_on_startup_cb (gpointer data)
 {
 	GsUpdateMonitor *monitor = data;
 
 	g_debug ("First hourly updates check");
-	check_updates (monitor);
-	get_upgrades (monitor);
+	restart_updates_check (monitor);
```

Comment:
This commit is already upstream. The reason I don't make dependable on allow-updates is that this is not really a check_updates, it's something more of a _refresh_and_eventually_get_updates_. Richard and I have agreed to rename gs-updates-monitor to gs-schedule which is less misguiding ATM but that will happen later.